### PR TITLE
PBI 2 - Add validator for JSON structure

### PR DESCRIPTION
**Pull request by NtapSlur**
Partially finished for PBI 2-4:
- Added method 'validate_json' to ensure the given JSON data 
- Testcase given:
1. Positive case: Valid JSON structure
2. Negative case: invalid JSON structure (including diagram = ClassDiagram, etc), empty JSON

### CODE REVIEW

File: app/parse_json_to_object_seq.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            data_json = json.loads(data)
+            
+            if(self.validate_json(data_json)):
+                self.__json = data_json
+                return "Success"
+
+            else:
+              raise Exception("Given .jet is not valid!")
+
+        except Exception:
```
Comment (MightyZanark) [inline app/parse_json_to_object_seq.py:20]: Too generic catch of an exception, try to see what exception might happen and catch them independently. For example JSONDecodeError. This is so we can log it appropriately as well.

**Comment by Not0nline** [comment]
untuk validate JSON apakah terdapat limit untuk storage yang dapat? apakah terdapat case dimana JSON file terlalu panjang dan terdapat bagian yang dicut atau juga storagenya terlalu penuh? apakah corner-corner case seperti ini sudah terpikirkan semuanya? 

Terima kasih
- Adrial

File: tests/test_parse_json_to_object_seq.py [hunk 2]
```diff
@@ -0,0 +1,463 @@
+import unittest
+import json
+from app.parse_json_to_object_seq import *
+
+class TestParseJsonToObjectSeq(unittest.TestCase):
+
+    def test_set_valid_json(self):
+        json_data = """
```
Comment (MightyZanark) [inline tests/test_parse_json_to_object_seq.py:8]: Might want to move the json test data to a json file instead of making a multiline string here.

**Review by MightyZanark — COMMENTED** [review]
Other than the ones I have commented on, looks good!

**Comment by DaWanAnOnli** [comment]
Great job!

From an architecture and best practice's point of view. Your code doesn't seem to contain any problems. Maybe something you should consider is putting a descriptive doc string for every class and method in your modifications. This will help with readability and scalability.
1. Readibility. This will help you or other programmers in the future to understand the code to make modifications or improvements.
2. Scalability. By having a well documented code, future developers will not be intimidated to expand the code for future plans. 

Ideally, each docstring contains sample input-output pair so that other developers know the expected input and output of the methods.

**Comment by KingOnyxx** [comment]
In my opinion, other than the comments everybody has left here, this already looks good to me. I can use this validate_json method in my work. You can start implementing the parse sequential diagram. Good luck!

**Comment by KenKomKom** [comment]
Based on what I understand, your test only handles negative case if it is an empty JSON and no other cases. I belief it is essential to make test cases to tests if the keys of the JSON are invalid as well as if the values in the JSON are invalid and the combination of both.

**Comment by NtapSlur** [comment]
> untuk validate JSON apakah terdapat limit untuk storage yang dapat? apakah terdapat case dimana JSON file terlalu panjang dan terdapat bagian yang dicut atau juga storagenya terlalu penuh? apakah corner-corner case seperti ini sudah terpikirkan semuanya?
> 
> Terima kasih
> 
> * Adrial

Terkait limit storage tidak dihandle dari validator JSON, yang notabene fungsi validator JSON adalah untuk memastikan bahwa struktrur dari JSON yang diberikan sesuai dengan struktur dari file .jet.

File: app/parse_json_to_object_seq.py (hunk 1 above)
Comment (NtapSlur) [reply app/parse_json_to_object_seq.py:20]: Thankyou for the feedback! Will revise this soon.

File: tests/test_parse_json_to_object_seq.py (hunk 2 above)
Comment (NtapSlur) [reply tests/test_parse_json_to_object_seq.py:8]: Thankyou for the suggestion!

**Comment by NtapSlur** [comment]
> Great job!
> 
> From an architecture and best practice's point of view. Your code doesn't seem to contain any problems. Maybe something you should consider is putting a descriptive doc string for every class and method in your modifications. This will help with readability and scalability.
> 
> 1. Readibility. This will help you or other programmers in the future to understand the code to make modifications or improvements.
> 2. Scalability. By having a well documented code, future developers will not be intimidated to expand the code for future plans.
> 
> Ideally, each docstring contains sample input-output pair so that other developers know the expected input and output of the methods.

Thankyou for the suggestion! I also think that doc string is important for better understanding of the code! Will add it!

**Comment by NtapSlur** [comment]
> Based on what I understand, your test only handles negative case if it is an empty JSON and no other cases. I belief it is essential to make test cases to tests if the keys of the JSON are invalid as well as if the values in the JSON are invalid and the combination of both.

Thankyou for the feedback! Gonna add the tests!
